Source/WebKit/UIProcess/API/glib/WebKitUserContentManager.cpp

```diff
@@ -224,12 +256,158 @@ void webkit_user_content_manager_remove_all_scripts(WebKitUserContentManager* ma
     manager->priv->userContentController->removeAllUserScripts();
 }
 
+/**
+ * WebKitUserScriptMessageReply: (ref-func webkit_user_script_message_reply_ref) (unref-func webkit_user_script_message_reply_unref)
+ *
+ * A helper class that sends user provided script message to the signal handler.
+ * If no reply has been sent by the user, an automatically generated reply with
+ * undefined value with be sent.
+ *
+ * Since: 2.40
+ */
+struct _WebKitUserScriptMessageReply {
+    _WebKitUserScriptMessageReply(WebCore::SerializedScriptValue& serializedScriptValue, WTF::Function<void(API::SerializedScriptValue*, const String&)>&& completionHandler)
+        : completionHandler(WTFMove(completionHandler))
+        , referenceCount(1)
+    {
+        JSCContext* jscContext = SharedJavascriptContext::singleton().getOrCreateContext();
+        JSValueRef jsValueRef = serializedScriptValue.deserialize(jscContextGetJSContext(jscContext), nullptr);
+        message = adoptGRef(jscValueCreate(jscContext, jsValueRef));
```

Comment:
We don't need to handle this here, the message can be a WebKitJavaScriptResult like the signal with no reply does.